### PR TITLE
add workspace switch

### PR DESCRIPTION
**Pull request by raintygao**
### Description

add workspace switch

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by raintygao** [comment]
This PR is a replacement for https://github.com/ruanyl/OpenSearch-Dashboards/pull/11, the original PR was closed due to change to use the new branch.

File: src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx [hunk 1]
```diff
@@ -50,17 +53,17 @@ export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
...
-      .finally(() => {
-        setLoading(false);
-      });
-  };
+  const onChange = useCallback(
+    (workspaceOption: WorkspaceOption[]) => {
+      /** switch the workspace */
+      setLoading(true);
+      const id = workspaceOption[0].key!;
+      workspaces?.client.enterWorkspace(id);
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx:61]: No need to call enterWorkspace here

File: src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx [hunk 2]
```diff
@@ -50,17 +53,17 @@ export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
...
+      workspaces?.formatUrlWithWorkspaceId(window.location.href, id);
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx:62]:
```suggestion
      window.location.href = workspaces?.formatUrlWithWorkspaceId(window.location.href, id);
```

File: src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx [hunk 3]
```diff
@@ -8,23 +8,26 @@ import React, { useState, useCallback, useMemo, useEffect } from 'react';
...
 }
 
 export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
-  const { coreStart, onCreateWorkspace, onSwitchWorkspace } = props;
+  const { coreStart, onCreateWorkspace } = props;
   const workspaceList = useObservable(coreStart.workspaces.client.workspaceList$, []);
   const currentWorkspaceId = useObservable(coreStart.workspaces.client.currentWorkspaceId$, '');
+  const {
+    services: { workspaces },
+  } = useOpenSearchDashboards();
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx:30]: different place to get the same reference `workspaces`. The approach to get `workspaces` client should be consistent in a single component.

File: src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx [hunk 4]
```diff
@@ -50,17 +49,19 @@ export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
...
-      });
-  };
+  const onChange = useCallback(
+    (workspaceOption: WorkspaceOption[]) => {
+      /** switch the workspace */
+      setLoading(true);
+      const id = workspaceOption[0].key!;
+      const newUrl = coreStart.workspaces?.formatUrlWithWorkspaceId(window.location.href, id);
+      if (newUrl) {
+        window.location.href = newUrl;
```
Comment (wanglam) [inline src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx:59]: What about using `navigateToUrl`? Is this API still need get current path?

Comment (raintygao) [reply src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx:59]: I tried to use `navigateToUrl`, but it seems invalid when calling switch in home.

**Review by wanglam — APPROVED** [review]

**Comment by raintygao** [comment]
@ruanyl  Could you give me a write access to this repo or help me merge it?

**Pull request merged**